README.md

```diff
@@ -1,87 +1,121 @@
+## Issues with using `form_tag`
 
-Problems with Edit Form
+Before we get into the benefits and features of the `form_for` method, let's first discuss some of the key drawbacks to utilizing `form_tag`:
 
-our form_tag isn't using any information about @post to make assumptions about our form action and method. @post.new_record? vs @post.persisted? If AR knows whether it's a new post or an existing post, and we have an existing post, isn't it obvious that the form's method should be PUT and the URL should be post_path for @post?
-repitition of @post and post everywhere. our entire form consitantly says that we're dealing with a post, both when we name a field, post_title, and then when we call a method on @post.
-repetition of attribute names in the input tags, we're naming the field after the same attribute we're reading to set the value of the field.
-Problems with Update
+* Our form has to manually be passed the route for where the form parameters will be submitted
 
-manual assignment is well manual. repitition of attribute we're writing too and the corresponding key in params, title= and params[:post_title]
-with update, we have to repeat the same information but now as keys in a hash. update(:title => params[:post_title]). No way to just pass params in because it has more information, such as ID, then we can mass assign. If only we could ask params for just the attributes relating to a post?
-The biggest offender though, is that this code violates a principal of beauty, proportion. Let's say we needed to add a new field to our form, like an attribute author_name. Conceptually, that's 1 change. How many changes do we need to make to our form and controller? We need to update the form and then we need to update the controller. One conceptual change should mean one code change. That would be a beatiful proportion.
+* The form has no knowledge of the form, it doesn't know if the form is meant to create or update a record
 
-form_for to the rescue
+* You're forced to have duplicate code throughout the form, it's hard to adhere to DRY principles when utilizing the `form_tag`
 
-form_tag family of helpers - simply generate html, no knowledge, not smart. helpful and shorter than html, but we can do better.
 
-form_for family of helpers are bound to the activerecord object the form relates too. that means every time we build a form tag or an input tag, we can make assumptions based on the model itself. it's really cool, watch.
+## Difference between `form_for` and `form_tag`
 
-form_for(@post) do |form_builder|
+The differences between `form_for` and `form_tag` are subtle, but important, below is a basic breakdown of the differences, we'll start with talking about them at a high level perspective and then get into each one of the aspects on a practical/implementation basis:
 
-end
+* `form_for` binds itself to the model and automatically auto fills the values for each attribute by simply calling the symbol
```

Comment:
"binds itself to the model" is a rough sentence. Maybe "The `form_for` method accepts the instance of the model as an argument. Using this argument, `form_for` is able to make a bunch of assumptions for you."